api/restHandler/AppListingRestHandler.go

```diff
@@ -144,23 +144,58 @@ func (handler AppListingRestHandlerImpl) FetchAppsByEnvironment(w http.ResponseW
 	t2 := time.Now()
 	handler.logger.Infow("api response time testing", "time", time.Now().String(), "time diff", t2.Unix()-t1.Unix(), "stage", "2")
 	t1 = t2
-	appEnvs := make([]*bean.AppEnvironmentContainer, 0)
-	rbacObjects := handler.enforcerUtil.GetRbacObjectsForAllApps()
-	for _, env := range envContainers {
-		if fetchAppListingRequest.DeploymentGroupId > 0 {
-			if env.EnvironmentId != 0 && env.EnvironmentId != dg.EnvironmentId {
-				continue
+
+	isActionUserSuperAdmin, err := handler.userService.IsSuperAdmin(int(userId))
+	if err != nil {
+		handler.logger.Errorw("request err, FetchAppsByEnvironment", "err", err, "userId", userId)
+		writeJsonResp(w, err, "Failed to check is super admin", http.StatusInternalServerError)
+		return
+	}
+	appEnvContainers := make([]*bean.AppEnvironmentContainer, 0)
+	if isActionUserSuperAdmin {
+		for _, envContainer := range envContainers {
+			appEnvContainers = append(appEnvContainers, envContainer)
+		}
+	} else {
+		var teamIds []*int
+		authorizedTeams := make(map[int]int)
+		for _, envContainer := range envContainers {
+			teamIds = append(teamIds, &envContainer.TeamId)
+		}
+		teams, err := handler.teamService.FindByIds(teamIds)
+		if err != nil {
+			handler.logger.Errorw("service err, FetchAppsByEnvironment", "err", err, "payload", fetchAppListingRequest)
+			writeJsonResp(w, err, "", http.StatusInternalServerError)
+		}
+		for _, team := range teams {
+			if ok := handler.enforcer.EnforceByEmail(userEmailId, rbac.ResourceTeam, rbac.ActionGet, team.Name); ok {
+				authorizedTeams[team.Id] = team.Id
+			}
+		}
+		filteredAppEnvContainers := make([]*bean.AppEnvironmentContainer, 0)
+		for _, envContainer := range envContainers {
+			if _, ok := authorizedTeams[envContainer.TeamId]; ok {
+				filteredAppEnvContainers = append(filteredAppEnvContainers, envContainer)
 			}
 		}
-		object := rbacObjects[env.AppId]
-		if ok := handler.enforcer.EnforceByEmail(userEmailId, rbac.ResourceApplications, rbac.ActionGet, object); ok {
-			appEnvs = append(appEnvs, env)
+		rbacObjects := make(map[int]string)
+		rbacObjects = handler.enforcerUtil.GetRbacObjectsForAllApps()
```

Comment:
we should get team name inside AppEnvironmentContainer and use it to generate RBAC object rather than making this call

Reply:
modified